js/modules/Forms/RendererController.js

```diff
@@ -127,39 +127,61 @@ export class GlpiFormRendererController
         $(this.#target).removeClass('pointer-events-none');
     }
 
-    #checkCurrentSectionValidity() {
-        // Find all required inputs that are hidden and not already disabled.
-        // They must be removed from the check as they are inputs from others
-        // sections or input hidden by condition (thus they should not be
-        // evaluated).
-        // The easiest way to not evaluate these inputs is to disable them.
-        const inputs = $(this.#target).find('[required]:hidden:not(:disabled)');
-        for (const input of inputs) {
-            input.disabled = true;
-        }
+    async #checkCurrentSectionValidity() {
+        // Get the UUID of the current section
+        const currentSectionElement = $(this.#target).find(`[data-glpi-form-renderer-section="${this.#section_index}"]`);
+        const currentSectionUuid = currentSectionElement.data('glpi-form-renderer-uuid');
+
+        const response = await $.ajax({
+            url: `${CFG_GLPI.root_doc}/Form/ValidateAnswers`,
+            type: 'POST',
+            data: `${$(this.#target).serialize()  }&section_uuid=${currentSectionUuid}`,
+            dataType: 'json',
+        });
 
-        // Check validity and display browser feedback if needed.
-        const is_valid = this.#target.checkValidity();
-        if (!is_valid) {
-            this.#target.reportValidity();
-        }
+        if (response.success === false) {
+            // Remove previous error messages
+            $(this.#target)
+                .find(".invalid-tooltip")
+                .remove();
+            $(this.#target)
+                .find(".is-invalid")
+                .removeClass("is-invalid");
+
+            Object.values(response.errors).forEach(error => {
+                // Highlight the field with error
+                const question = $(`[data-glpi-form-renderer-id="${error.question_id}"][data-glpi-form-renderer-question]`);
+                if (question.length) {
+                    // Find the input field within the question
+                    const inputField = question.find('input:not([data-uploader-name]), select, textarea');
+                    if (inputField.length) {
+                        // Generate a unique ID for the error message
+                        const errorId = `error-${error.question_id}`;
+
+                        // Add validation classes and accessibility attributes
+                        inputField
+                            .addClass('is-invalid')
+                            .attr('aria-invalid', 'true')
+                            .attr('aria-errormessage', errorId);
+
+                        // Add a tooltip with the error message
+                        inputField.parent().append(
+                            `<span id="${errorId}" class="invalid-tooltip">${error.message}</span>`
+                        );
+                    }
+                }
```

Comment:
We could use early returns to reduce nesting (`return` works as a `continue` in `forEach`.

```suggestion
                if (!question.length) {
                    return;
                }
                
                // Find the input field within the question
                const inputField = question.find('input:not([data-uploader-name]), select, textarea');
                if (!inputField.length) {
                    return;
                }
                
                // Generate a unique ID for the error message
                const errorId = `error-${error.question_id}`;

                // Add validation classes and accessibility attributes
                inputField
                    .addClass('is-invalid')
                    .attr('aria-invalid', 'true')
                    .attr('aria-errormessage', errorId);

                // Add a tooltip with the error message
                inputField.parent().append(
                    `<span id="${errorId}" class="invalid-tooltip">${error.message}</span>`
                );
```